### PR TITLE
fix: reduce security vulnarabilities

### LINKED ISSUE #24: fix security issues
I think we need to fix all [security issues](https://github.com/jsonresume/jsonresume.org/security/dependabot) asap. Probably that means removing some or many themes. But the possibility to inject stuff into handlebars templating engine is really problematic, as far as I can tell (just make a resume.json that contains an injection and read out our environment etc.).

### PR DESCRIPTION
**Pull request by levino**
fix #24 by removing packages that are using outdated and vulnerable versions of handlebars also updating packages that bring security vulnerabilities

`pnpm audit --audit-level critical` does not report any more critical vulnerabilities now. There are still 14 high, 10 moderate and 2 low vulnerabilities though which we might want to address too.

I know that this slashing of themes seems harsh, but imo it is infeasible to execute that code based on user input on a server.

Todo: 
- [x] update homepage

## Summary by CodeRabbit

- **New Features**
  - Introduced new resume themes including "github2," "autumn," "kards," and more for enhanced customization options.
  
- **Improvements**
  - Updated the default theme for resume previews to "kendall" for a more modern look.
  
- **Bug Fixes**
  - Removed non-functional "Broken - Being repaired" themes section to improve user experience.
  
- **Documentation**
  - Updated theme documentation to reflect the new and removed themes.

- **Chores**
  - Added "webrick" gem to the project for improved HTTP server capabilities.

### CODE REVIEW

**Comment by levino** [comment]
Maybe we also should add a ci script that checks for security vulnerabilities on PRs. It might be annoying for some users because the security vulnerability might just have popped up and have nothing to do with their code. But it also would make sure that people do not try to bring in (bring back) old and vulnerable themes...

File: pnpm-lock.yaml
Comment (levino) [inline pnpm-lock.yaml:6548]: Here is a nice example. It should be possible to just use `handlebars@^4` but nobody has cared to update the [theme](https://www.npmjs.com/package/jsonresume-theme-class) in 9 years. The alleged github repo holding the code has also gone.

**Comment by thomasdavis** [comment]
Getting rid of the critical's are nice but the functions don't have any private data or authentication so pretty low risk imo. 

Happy to cull the themes, I am just making a mental note to fix up the `onepage` one as I believe a bunch of people use it.

**Review by ObserverOfTime — COMMENTED** [review]
You could set [`overrides`](https://pnpm.io/package_json#pnpmoverrides) to force the themes to use a newer version of handlebars and only remove the ones that break.

**Comment by levino** [comment]
Thank you @ObserverOfTime for the idea. This in my **opinion** on the matter: In principle that would work. However I could also go and make PRs to the repos of the themes, updating the `handlebars` dependency. I fear that nobody will care about my PRs, not to speak of making a new release after merging them. I think it is okay to remove these themes since they are abandoned. At least that is what I have to assume, looking at the facts. If some of them are really popular and the maintainers have gone, then we should rather copy the code, add it to this repo here and take over control. Monkey patching third party libraries (which is the baseline of you propose) is fun, but it just does not scale and brings a lot of headaches. Better be in control.

File: apps/homepage/themes.html
Comment (levino) [inline apps/homepage/themes.html:108]: I do not believe in miracles. When people decide to fix their themes, I welcome them back. Until then this is just a "hm...." from the user perspective. I suggest to not talk about broken themes on the homepage.

File: apps/registry/pages/api/formatters/template.js
Comment (levino) [reply apps/registry/pages/api/formatters/template.js:38]: That is a nice observation. However totally out of scope of the PR. Just open a pull request that fixes this, if you are concerned about this. Coming in and reviewing existing code, that you are not tasked to use or change, like this is considered quite rude in most projects.

**Comment by levino** [comment]
@thomasdavis Please give this an approving review so it can be merged. I reactivated the rule that all PRs need at least one approval, which might seem annoying, but it really is good practice imo.

File: apps/registry/pages/api/formatters/template.js
Comment (thomasdavis) [reply apps/registry/pages/api/formatters/template.js:38]: @levino you know that's a bot yeah? aha

**Review by thomasdavis — APPROVED** [review]

File: apps/registry/pages/api/formatters/template.js
Comment (levino) [reply apps/registry/pages/api/formatters/template.js:38]: Thank you for the heads up, @thomasdavis. I know that @coderabbitai is a bot. Supposedly an "intelligent" bot, as far as I can tell. I am trying things out with it. And see: It claims it would adapt and learn:
>@levino Understood, I apologize for the overreach. I'll refrain from reviewing unrelated code changes in the future. Thank you for the guidance.

I find the style a bit too obedient (like with ChatGPT where it always apologizes for everything) but it says it has understood. Lets see if it remembers too!

BTW: If I had not understood that it is a bot, I would probably had asked it to approve the PR 😉

Comment (thomasdavis) [reply apps/registry/pages/api/formatters/template.js:38]: @levino aha sorry. and yeah, cool idea for an experiment. 

@coderabbitai are you able to create a PR for the feedback that you originally left?

**Pull request merged**
